django/contrib/admin/helpers.py

```diff
@@ -214,6 +232,9 @@ def contents(self):
             else:
                 if isinstance(f.remote_field, ManyToManyRel) and value is not None:
                     result_repr = ", ".join(map(str, value.all()))
+                elif (isinstance(f.remote_field, ForeignObjectRel) or isinstance(f.remote_field, OneToOneField)) \
+                        and value is not None:
```

Comment:
```suggestion
                elif (
                    isinstance(f.remote_field, (ForeignObjectRel, OneToOneField)) and
                    value is not None
                ):
```

Reply:
I think this should take into account the `parent_link` attribute for `OneToOneField`s :thinking:

Reply:
Thank you very much for the review. I'll fix the other issues, but I'm not sure what is the expected behavior for `parent_link`:

It seems if I have the following
```
class ModelParent(models.Model):
    name = models.CharField(max_length=60)

class ModelChild(ModelParent):
    parent = models.OneToOneField(ModelParent, models.CASCADE, primary_key=True, parent_link=True)
```

The the behavior of the admin on master is to not display `parent` and instead display the parent's `name` directly as a field in the admin form for `ModelChild`.

So I feel there isn't anything special to do for `OneToOneField` with `parent_link=True` since those don't seem to be displayed anyway.

I might very well be missing something and I have to say I'm not super familiar with `parent_link=True`, so maybe I misunderstand the expected behavior. What did you have in mind for the behavior with `parent_link` ?

Reply:
True, I was thinking that maybe it's worth to cover them in tests, but you're right be don't need to test them separately.

Reply:
Ok thanks !